src/modules/claims/claims.service.ts

```diff
@@ -545,12 +545,16 @@ export class ClaimsService {
       return;
     }
 
-    if (!subjectAgreement && subject === this._signerService.did) {
+    if (!subjectAgreement) {
+      // If the signer is the subject, use the signer DID. If the signer is not the subject, use the subject DID:
+      const subjectAgreer =
+        subject === this._signerService.did ? this._signerService.did : subject;
       subjectAgreement = await this.approveRolePublishing({
-        subject: this._signerService.did,
```

Comment:
@jrhender Thanks for the input on this. Is this what you were thinking? There is no concept in the claim of this being an asset, other than that the subject is different than signer. Is this logic what you were thinking? This causes an error in Claim Manager - see test case above (VM Exception while processing transaction: revert ClaimManager: agreement signer is not authorized to sign on behalf of subject), but maybe what you were getting at is in either case, the subject needs to be the signer service (although I don't think this is what you were getting at, as it wouldn't be true to the fact that the signer is not the subject of the claim).

Reply:
@jrhender would it be safe to say that the subject is always the subject? I think the logic check is unnecessary, right? If the signer is the subject, or its not and you use the subject, you are always just using the subject. So it would be:

```
    if (!subjectAgreement) {
      // If the signer is the subject, use the signer DID. If the signer is not the subject, use the subject DID:
      subjectAgreement = await this.approveRolePublishing({
        subject: subject,
        role: claimType,
        version: +claimTypeVersion,
      });
      // }
    }
```
Is this right?

Reply:
Is this right though, in the case of assets, can an asset be a subject that is agreeing to something, if it is not a real DID with signing capabilities?

Reply:
> There is no concept in the claim of this being an asset, other than that the subject is different than signer.

Yes, this is right

> would it be safe to say that the subject is always the subject?

Yes, I think so

> Is this right though, in the case of assets, can an asset be a subject that is agreeing to something, if it is not a real DID with signing capabilities

Well, the owner can update the DID document to add a key that can sign on behalf of the asset DID